forms/Form.php

```diff
@@ -1415,9 +1415,10 @@ public function loadDataFrom($data, $mergeStrategy = 0, $fieldList = null) {
 				// If field is in array-notation we need to access nested data
 				else if(strpos($name,'[')) {
 					// First encode data using PHP's method of converting nested arrays to form data
-					$flatData = urldecode(http_build_query($data));
+					$flatData    = http_build_query($data);
+					$encodedName = urlencode($name);
```

Comment:
Hey Luke, great to hear from you! Stylistically, I think declaring a one-off variable here is adding noise rather than clarity.